src/Data/Algorand/Transaction.hs

```diff
@@ -349,3 +352,13 @@ instance ToJSON StateSchema where
 
 instance FromJSON StateSchema where
   parseJSON = parseCanonicalJson
+
+decodeUintAppArgument :: ByteString -> Maybe Word64
```

Comment:
Could you please add comments explaining the purpose of these new functions? It's not obvious from their names/definitions.